xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/migration/R141004000XWIKI20575DataMigration.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.search.solr.internal.migration;
+
+import java.io.IOException;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.solr.client.solrj.SolrServerException;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.search.solr.internal.api.SolrInstance;
+
+import com.xpn.xwiki.store.migration.DataMigration;
+import com.xpn.xwiki.store.migration.DataMigrationException;
+import com.xpn.xwiki.store.migration.XWikiDBVersion;
+
+/**
+ * Migration in charge of emptying the Search solr core in order to perform a reindex of all documents after a
+ * regression has been fixed.
+ *
+ * @version $Id$
+ * @since 15.0
+ * @since 14.10.4
+ */
+@Component
+@Named(R141004000XWIKI20575DataMigration.HINT)
+@Singleton
+public class R141004000XWIKI20575DataMigration implements DataMigration
```

Comment:
Shouldn't this implement `HibernateDataMigration` because of https://jira.xwiki.org/browse/XWIKI-19399?

Reply:
Yes I was planning to change that as it was discussed on the chat. I will also add a FIXME comment in DataMigration interface since it's easy to forget this issue...